src/helpers/in-memory-cache.ts

```diff
@@ -0,0 +1,48 @@
+import type { VirtualCurrencies } from "../entities/virtual-currencies";
+
+interface CacheEntry<T> {
+  data: T;
+  timestamp: number;
+}
+
+export class InMemoryCache {
+  private virtualCurrenciesCache = new Map<
+    string,
+    CacheEntry<VirtualCurrencies>
+  >();
+  readonly CACHE_EXPIRY_MS = 5 * 60 * 1000; // 5 minutes
+
+  cacheVirtualCurrencies(
+    appUserID: string,
+    virtualCurrencies: VirtualCurrencies,
+  ): void {
+    this.virtualCurrenciesCache.set(appUserID, {
+      data: virtualCurrencies,
+      timestamp: Date.now(),
+    });
+  }
+
+  getCachedVirtualCurrencies(appUserID: string): VirtualCurrencies | null {
+    const entry = this.virtualCurrenciesCache.get(appUserID);
+    return this.getCachedData(entry ?? null);
+  }
+
+  invalidateVirtualCurrenciesCache(appUserID: string): void {
```

Comment:
I guess it shouldn't be an issue by having the map keyed by user ID... But I wonder if we should clear the cache when changing users... More of a cleanup note. We also clear caches when changing users on mobile and seems like a good practice...

Reply:
Good idea! If we decide to go forward with the caching, I'll update the PR to clear the cache out when `changeUser()` is called 👍